src/site/markdown/server/manage-guice-distributed-james.md

```diff
@@ -205,3 +206,61 @@ reprocessing all the failed events registered in event dead letters.
 If for some other reason you don't need to redeliver all events, you have more fine-grained operations allowing you to
 [redeliver group events](manage-webadmin.html#Redeliver_group_events) or even just
 [redeliver a single event](manage-webadmin.html#Redeliver_a_single_event).
+
+## Solving cassandra inconsistencies
+
+Many tables in James have close relationships. Some operation requires cassandra writes to those tables, 
+if there is any failure upon in the middle of an operation. It could bring an inconsistent state to the data.
+
+Because of the lack of transaction, it's hard to prevent these kind of issues. We had developed some features to 
+fix some existing cassandra inconsistency issues had been reported to James. 
+
+There are typical type inconsistencies:
+ - [RRT(RecipientRewriteTable) mapping sources](#rrtrecipientrewritetable-mapping-sources)
+ - [Jmap message fast view projections](#jmap-message-fast-view-projections)
+ - [Mailboxes](#mailboxes)
+
+### RRT(RecipientRewriteTable) mapping sources
+
+`rrt` and `mappings_sources` tables store information about address mappings. 
+While `rrt` is the source of truth and `mappings_sources` is the projection table containing all 
+mapping sources.
+
+#### How to detect the inconsistencies
+
+// TODO
+
+#### How to solve
+
+Execute the Cassandra mapping `SolveInconsistencies` task describing in [webadmin documentation](#https://james.apache.org/server/manage-webadmin.html#Operations_on_mappings_sources) 
+
+### Jmap message fast view projections
+
+When you read a Jmap message, some calculated properties are expected to be fast to retrieve, like `preview`, `hasAttachment`. 
+By pre calculate and store them into a message projection table(`message_fast_view_projection`). The fetch operation is optimized in 
```

Comment:
missing space

Reply:
There's a space, you just can't see it via this interface :)